lib/encoder_impl.cc

```diff
@@ -286,6 +287,12 @@ void encoder_impl::set_ps(std::string ps) {
 
 		std::memset(PS, ' ', sizeof(PS));
 		std::memcpy(PS, ps.c_str(), len);
+
+        using std::cout;
+        using std::endl;
+        cout << "PS set to \"" << ps << "\"" << endl;
+        rebuild();
+        rebuild();
```

Comment:
i won't start a discussion about tabs vs. spaces in a comment section on github, but I hope you agree that mixing a lot of very different indentation styles within one file doesn't make much sense.